WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Helpers/DashboardPostsSyncManager.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+protocol DashboardPostsSyncManagerListener: AnyObject {
+    func postsSynced(success: Bool, blog: Blog, posts: [AbstractPost]?, for statuses: [String])
+}
+
+class DashboardPostsSyncManager {
+
+    // MARK: Type Aliases
+
+    typealias SyncSuccessBlock = () -> Void
+    typealias SyncFailureBlock = (Error?) -> Void
+
+    // MARK: Private Variables
+
+    private let postService: PostService
+    private let blogService: BlogService
+    private let lock = NSRecursiveLock()
```

Comment:
This lock isn't being used anywhere 👀 

```suggestion
```

Reply:
Nice catch! I was using this before going with the atomic property wrapper.
Removed in 6ffde5f9779d1768e9d2da50676cf5f313687ca7